gemrb/GUIScripts/pst/GUIREC.py

```diff
@@ -1188,15 +1188,29 @@ def GetThac0 (Class, Level):
 	return Thac0Table.GetValue (Class, str (Level))
 
 #apparently the original code doesn't follow profsmax/profs tables
-def HasGainedWeapProf (pc, currProf, currLevel, Class):
-	#only fighters gain weapon proficiencies
-	if Class!="FIGHTER":
-		return False
-	#hardcoded limit is 4
-	if currProf>3:
+def HasGainedWeapProf (pc, currProf, currLevel):
+
+ 	Class = GUICommon.GetClassRowName(pc)
+
+	# Limits obtained from testing original game
+	limit = 1000
+	if Class == "FIGHTER":
+		# Morte, Nordom amd Vhailor are limited to 5, but not TNO
+		if not GUICommon.NamelessOneClass(pc):
+			limit = 5
+	elif Class == "FIGHTER_MAGE" or Class == "FIGHTER_THIEF":
+		# Dakkon and Annah are limited to 4
+		limit = 4
+	else:
```

Comment:
What about Ignus the mage?

Reply:
That's a good point, I've never really included Ignus in my party so I can't remember, but it could be possible he is meant to have some progression. Easy to test though.

Reply:
Just checked, even at level 20 he only has one prof point and a thac0 of 14

Reply:
It seems to be a little bit arbitrary, just drew this up for reference:
```
Dakkon
Fighter		4	5	5	6	6	6	7	7	8	8	9	9	10	11
Mage		4	4	5	5	6	7	7	8	8	9	10	11	11	12
Points		1	2	2	2	2	2	2	2	3	3	3	3	3	4
Special	Blade shift at level 7 (+2), 10 (+3)

Annah
Fighter		4	5	5	5	6	6	7	8	8	8	9	9	10	10	11
Thief		4	4	5	6	6	7	7	9	10	11	11	12	12	13	13
Points		1	2	2	2	2	2	2	3	3	3	3	3	3	3	4

Morte
Fighter		2	3	4 	5	6	7	8	9	10	11
Points		2	2	2	3	3	3	4	4	4	5

Vhailor
Fighter		9	10	11	12	13	14	15	16	17
Points		2	2	3	3	4	4	4	4	5

Nordom
Fighter		6	7	8 	9	10	11	12	13	
Ponts		2	2	3	3	3	4	4	5
```

Dakkon and Annah get a point when their fighter level hits 5, 8, 11, capped at 4
They do only get points for fighter levels, but the progression is moved ahead one level so they get them sooner

Morte starts with 2 points, and also gets one at 5,8,11, capped at 5

Vhailor starts with 2 at level 9, gets a point at 11, 14 then 17 capped at 5

Then Nordom spoils the pattern by getting a point at lv 13 instead of 14 :)

Grace and Ignus get nothing. You'd think Ignus would practice throwing his fireballs just for the fun of it...

TNO starts with 2 spent and 3 unspent points, gains one at level 6 and every three fighter levels thereafter, regardless of how many levels in other classes. However his ability to spend proficiencies in a particular weapon is capped according to his class specialisation, I believe.

Reply:
A table is the way to go then and it will nicely encode at least FFG. Are you saying TNO doesn't get any extra points by just maxing out a thief or a mage?

Reply:
No, you can be level 125 mage or thief and still get nothing

Interestingly though, for TNO his proficiency progression continues after level 20, seemingly still 1 point every three levels. I racked it up to 34 unspent points and stopped testing